docs/src/rfc/text/0000-core-static.md

```diff
@@ -0,0 +1,307 @@
+
+impl <P: Platform> Core {
+    fn set_instance(&'static self) {
+        let ptr = NonNull::from(self).cast::<u8>().as_ptr();
+        __SELF.store(ptr, core::sync::atomic::Ordering::SeqCst);
+    }
+
+    pub(crate) fn instance<'a>() -> &'a Self {
+        let ptr = __SELF.load(core::sync::atomic::Ordering::SeqCst);
+        unsafe {
+            NonNull::new(ptr).expect("Core instance is already initialize")
+                .cast::<Self>()
+                .as_ref()
+        }
+    }
+    pub fn init_memory(&'static self, physical_hob_list: *const c_void) -> &'static self {
+        self.set_instance();
+
+        ...
+    }
+}
+```
+
+The next final tidbit which is handy for this implementation is that `impl`'s of a struct do not have to be in the same
+module that the struct is defined. This allows us to scatter `Core<P>` impls throughout the workspace.
+
+A short and sweet example of how we will implement efiapi functions across the workspace is seen below:
+
+```rust
+impl <P: Platform> Core {
+    pub(crate) fn trust(handle: efi::Handle, file: &efi::Guid) -> Result<(), EfiError> {
+        let dispatcher = self.uefi_state.dispatcher_context.lock();
+        for driver in dispatcher.pending_drivers.iter_mut() {
+            if driver.firmware_volume_handle == handle && OrdGuid(driver.file_name) == OrdGuid(*file) {
+                driver.security_status = efi::Status::SUCCESS;
+                return Ok(());
+            }
+        }
+        Err(EfiError::NotFound0)
+    }
+
+    extern "efiapi" fn trust_efiapi(firmware_volume_handle: efi::Handle, file_name: *const efi::Guid) -> efi::Status {
+        if file_name.is_null() {
+            return efi::Status::INVALID_PARAMETER;
+        }
+
+        let file_name = unsafe { file_name.read_unaligned() };
+
+        match Self::instance().trust(firmware_volume_handle, &file_name) {
+            Err(status) => status.into(),
+            Ok(_) => efi::Status::SUCCESS,
+        }
+    }
+
+    fn init_dxe_services(system_table: &mut EfiSystemTable) {
+        ...
+        trust: Self::trust_efiapi,
+    }
+}
+```
+
+### Platform Interface changes
+
+We've now seen all of the core changes. What does the the change look for the platform? The platform now manually
+implements the `Platform` trait. Pretty much all platform config has moved there. That is about it. Lets see the
+example:
+
+```rust
+struct Platform;
+
+impl patina::Platform for Platform {
+    const PRIORITIZE_32_BIT_MEMORY: bool = false;
+
+    type Extractor: SectionExtractor = CompositeSectionExtractor;
+
+    fn section_extractor -> Self::Extractor { CompositeSectionExtractor::default() }
+
+    fn components(adder: &mut Adder<Component>) {
+        adder.add_component(...);
+        adder.add_component(...);
+    }
+
+    fn configs(adder: &mut Adder<Config>) {
+        adder.add_config(...);
+    }
+
+    fn services(c: &mut Addr<Service>) {
+        adder.add_service(...);
+    }

```

Comment:
Pedantry: I think these are inconsistent with the platform trait described previously (e.g. `Adder<Config>` vs. `Add<Config>`)